### PR TITLE
Add data size benchmark: monitor size of file

### LINKED ISSUE #913: Monitor and benchmark ICU4X data size?
Is it worthwhile benchmarking/monitoring the ICU4X data size? Pros and cons.
Con: The data is what it is, and sending alerts when the size increases may be an unwelcome distraction most of the time.
Pro: It can make it easier to find out and explain when the data size increases and why.

### PR DESCRIPTION
**Pull request by gnrunge**
provider/testdata/data/testdata.postcard.

Fixes #913

### CODE REVIEW

**Review by Manishearth — DISMISSED** [review]

**Comment by Manishearth** [comment]
Would also like at least one other review

**Review by gregtatum — APPROVED** [review]
Seems reasonable to me.

File: .github/workflows/build-test.yml [hunk 1]
```diff
@@ -818,6 +817,69 @@ jobs:
...
+        cargo run --package icu_benchmark_binsize  -- provider/testdata/data/testdata.postcard file | tee benchmarks/binsize/datasize/output.txt
+
+    - name: Store benchmark result (non-merge action only)
+      # Data from anything that is not a merge to mainline goes to a preliminary branch
+      if: github.event_name != 'push' || github.ref != 'refs/heads/main'
+      # Use gregtatum special feature to process ndjson-formatted benchmark data
+      uses: gregtatum/github-action-benchmark@d3f06f738e9612988d575db23fae5ca0008d3d12
+      with:
+        tool: 'ndjson'
+        output-file-path: benchmarks/binsize/datasize/output.txt
```
Comment (echeran) [inline .github/workflows/build-test.yml:842]: nitpick(non-blocking): Remove the `binsize` path segment from `benchmarks/binsize/datasize/...` here and below.  

It makes it seem like the data size testing is a sub-concept of binary size testing. I think of binary size testing as measuring the size of the compiled code only.

**Review by echeran — APPROVED** [review]

**Review by echeran — APPROVED** [review]

**Review by sffc — APPROVED** [review]
thanks!

**Pull request merged**